python/lsst/daf/butler/registry/_registry.py

```diff
@@ -122,19 +126,70 @@ class Registry:
     """
 
     @classmethod
-    def fromConfig(cls, config: Union[ButlerConfig, RegistryConfig, Config, str], create: bool = False,
+    def createFromConfig(cls, config: Optional[Union[RegistryConfig, str]] = None,
+                         dimensionConfig: Optional[Union[DimensionConfig, str]] = None,
+                         butlerRoot: Optional[str] = None) -> Registry:
+        """Create registry database and return `Registry` instance.
+
+        This method initializes database contents, database must be empty
+        prior to calling this method.
+
+        Parameters
+        ----------
+        config : `RegistryConfig` or `str`, optional
+            Registry configuration, if missing then default configuration will
+            be loaded from registry.yaml.
+        dimensionConfig : `DimensionConfig` or `str`, optional
+            Dimensions configuration, if missing then default configuration
+            will be loaded from dimensions.yaml.
+        butlerRoot : `str`, optional
+            Path to the repository root this `Registry` will manage.
```

Comment:
URI to repository root, not a path (I see that fromConfig is now wrong).

Reply:
Hmm, in this case `butlerRoot` is only used as an argument to `RegistryConfig.replaceRoot()`. I'm not sure that we can use generic URI with something like `sqlite://<butlerRoot>/gen3.sqlite3`?